### PR TITLE
improve thread safety for concurrent tiff loads

### PR DESCRIPTION
**Pull request by angusdavisadobe**
## Description
This fixes a thread-safety issue with concurrent parsing of multiple tiff files. The code uses several static std::unordered_maps and the indexing operator[] was used to perform the lookup. Use of this operator has the side effect of adding a default value to the map if the key was not found. This is not what should happen and isn't thread safe. The proper fix is to make the maps const and use the find() method of the map to perform the lookup and return a default value if not found.

## Tests

Tests were not performed. However the code changes were tested within the context of loading USD files with multiple tiff files.

## Checklist:

- [X ] I have read the [contribution guidelines](https://github.com/OpenImageIO/oiio/blob/master/CONTRIBUTING.md).
- [ ] If this is more extensive than a small change to existing code, I
  have previously submitted a Contributor License Agreement
  ([individual](https://github.com/OpenImageIO/oiio/blob/master/src/doc/CLA-INDIVIDUAL), and if there is any way my
  employers might think my programming belongs to them, then also
  [corporate](https://github.com/OpenImageIO/oiio/blob/master/src/doc/CLA-CORPORATE)).
- [ ] I have updated the documentation, if applicable.
- [ ] I have ensured that the change is tested somewhere in the testsuite
  (adding new test cases if necessary).
- [X ] My code follows the prevailing code style of this project.

### CODE REVIEW

File: src/libOpenImageIO/icc.cpp [hunk 1]
```diff
@@ -131,20 +137,32 @@ icc_color_space_name(string_view color_space)
...
     };
-    return primary_platforms[platform];
-}
+    // std::unordered_map::operator[](const Key& key) will add the key to the map if
+    // it doesn't exist. This isn't what is intended and isn't thread safe.
+    // Instead, just do the lookup and return the value or a nullptr.
+    //
+    // return primary_platforms[platform];
+    auto it = primary_platforms.find(platform);
+    return (it != primary_platforms.end()) ? it->second : nullptr;
```
Comment (lgritz) [inline src/libOpenImageIO/icc.cpp:165]: I think you lost a `}` at this line...

File: src/libOpenImageIO/icc.cpp [hunk 2]
```diff
@@ -131,20 +137,32 @@ icc_color_space_name(string_view color_space)
...
     };
-    return color_space_names[color_space];
+    // std::unordered_map::operator[](const Key& key) will add the key to the map if
+    // it doesn't exist. This isn't what is intended and isn't thread safe.
+    // Instead, just do the lookup and return the value or a nullptr.
+    //
+    // return color_space_names[color_space];
+    auto it = color_space_names.find(color_space);
+    return (it != color_space_names.end()) ? it->second : nullptr;
+
```
Comment (lgritz) [inline src/libOpenImageIO/icc.cpp:147]: Can you get rid of the extra blank line here, please?

**Review by lgritz — COMMENTED** [review]
Wow, good catch. It did not occur to me that it might be queried with a key that wasn't already in the map, thus potentially altering it and breaking thread safety. I should have declared these const all along.

The whole thing LGTM except for the syntax error I pointed out with the missing brace.

File: src/libOpenImageIO/icc.cpp (hunk 1 above)
Comment (angusdavisadobe) [reply src/libOpenImageIO/icc.cpp:164]: Sorry for the carelessness. I added back the missing `}`

File: src/libOpenImageIO/icc.cpp (hunk 2 above)
Comment (angusdavisadobe) [reply src/libOpenImageIO/icc.cpp]: I removed the blank line.

**Review by lgritz — APPROVED** [review]
LGTM, thanks!

**Pull request merged**